src/scripts/domParsers/khanAcademyVideoExtractor.ts

```diff
@@ -0,0 +1,50 @@
+import {VideoExtractor} from "./videoExtractor";
+import {VideoUtils} from "./videoUtils";
+import {YoutubeVideoExtractor} from "./youtubeVideoExtractor";
+
+import {Utils} from "../utils";
+
+export class KhanAcademyVideoExtractor implements VideoExtractor {
+	private youtubeExtractor: YoutubeVideoExtractor;
+
+	constructor() {
+		this.youtubeExtractor = new YoutubeVideoExtractor();
+	}
+
+	public getVideoIds(pageUrl: string, pageContent: string): string[] {
+		let regex = /id\s*=\s*("\s*video_(\S+)\s*"|'\s*video_(\S+)\s*')/gi;
+		let regexTwo = /data-youtubeid\s*=\s*("\s*(\S+)\s*"|'\s*(\S+)\s*')/gi;
+		let regexes = [regex, regexTwo];
+		return VideoUtils.matchRegexFromPageContent(pageContent, regexes);
+	}
+
+	public getVideoSrcValues(pageUrl: string, pageContent: string): string[] {
+		if (Utils.isNullOrUndefined(pageContent)) {
+			return;
+		}
+
+		let videoIds = this.getVideoIds(pageUrl, pageContent);
+		if (Utils.isNullOrUndefined(videoIds)) {
```

Comment:
You could collapse these two if statements (the null/undefined check and the length check)